File: 1_50/7_reverse_integer.java
class Solution {
    public int reverse(int x) {
        int sign = x > 0 ? 1 : -1;
        x = x > 0 ? x : -x;
        long rev = 0;
            
        while (x != 0) {
            rev = rev * 10 + (long) x % 10;
            
            // check if reversed overflows
            if (rev != (int) rev) {
                return 0;
            }
            
            x /= 10;
        }
        return (int) rev * sign;
    }
}

File: 1_50/7_reverse_integer.py

class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        rev = ""
        if x >= 0:
            rev = int(str(x)[::-1])
        else:
            rev = int(str(x)[:0:-1]) * -1
        
        # check if reversed overflows
        return rev if abs(rev) < 2 ** 31 else 0
